webapp/channels/src/actions/views/channel.ts

```diff
@@ -166,19 +166,19 @@ export function leaveChannel(channelId: string): ActionFuncAsync {
         const teamUrl = getCurrentRelativeTeamUrl(state);
 
         if (!isArchivedChannel(channel)) {
-            LocalStorageStore.removePreviousChannel(currentUserId, currentTeam.id, state);
+            LocalStorageStore.removePreviousChannel(currentUserId, currentTeam.id, state as GlobalState);
```

Comment:
Why is this being casted multiple times? Should the above `const state = getState() as GlobalState;` not be enough?

Reply:
Let me see if I can put the state assertion once, but maybe it is problematic, because some things get one GlobalState and others take the other one.